app/controllers/collections_controller.rb

```diff
@@ -115,6 +117,19 @@ def create_update_proposal
     redirect_to collection_draft_proposal_path(proposal)
   end
 
+  def loss_report
+    # When a user wants to use MMT to edit metadata that currently exists in a non-UMM form,
+    # it's important that they're able to see if any data loss occurs in the translation to umm.
+    # This method is needed to reference the appropriate helper and view for the lossiness report.
+    # If compared_collections is false, the error message will appear. Note that compared_collections
+    # is false when the cmr calls aren't successful.
+    compared_collections = prepare_translated_collections
+    respond_to do |format|
+      format.text { render plain: JSON.pretty_generate(@collection) + (compared_collections ? loss_report_output(compared_collections, hide_items: true, display: 'text') : 'Failure to get_concept or translate_collection' )}
```

Comment:
I think that these blocks are quite long one liners, and they are important/complicated enough that it is worth splitting them out into `if/else` statements